### PR TITLE
Secondary mechanisms

### LINKED ISSUE #13: Make a class for the alternative hatch system
Has at least 2 independent pistons (Solenoid): one to extend, one to expand.

### LINKED ISSUE #14: Develop a way to switch between using the 2 hatch classes easily
For example, using a big if/else block based on `boolean usesVelcroArm`.

### LINKED ISSUE #17: Write code for aiming light (Hatch)

### PR DESCRIPTION
**Pull request by TaeSeanDo**
Includes ExpanderHatch and a way between the 2 mechanisms (`boolean usesVelcroHatch`), along with aiming lights and cameras. Closes #17, closes #14, and closes #13.

### CODE REVIEW

File: src/main/java/com/teamcautionrobotics/robot/AimingLights.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+package com.teamcautionrobotics.robot;
+
+import edu.wpi.first.wpilibj.*;
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/AimingLights.java]: Avoid using `import *;` and instead name the classes being imported.

File: src/main/java/com/teamcautionrobotics/robot/ExpanderHatch.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
+package com.teamcautionrobotics.robot;
+
+import edu.wpi.first.wpilibj.*;
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/ExpanderHatch.java]: Avoid using `import *;` and instead name the classes being imported.

File: src/main/java/com/teamcautionrobotics/robot/ExpanderHatch.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+public class ExpanderHatch{
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/ExpanderHatch.java]: Add a space before the bracket.
```suggestion
public class ExpanderHatch {
```

File: src/main/java/com/teamcautionrobotics/robot/Robot.java [hunk 4]
```diff
@@ -25,131 +26,154 @@
...
-    //if B is pressed, deploy hatch pneumatics
+    VelcroHatch velcroHatch;
+    ExpanderHatch expanderHatch;
+    double armPower;
+    EnhancedJoystick driverLeft, driverRight;
+    Gamepad manipulator;
+    DriveBase driveBase;
+    AimingLights aimingLights;
+    double driveLeftCommand;
+    double driveRightCommand;
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/Robot.java:42]: Try to sort these and make them into smaller groups (example below).
```java
EnhancedJoystick driverLeft, driverRight;
Gamepad manipulator;

DriveBase driveBase;

double armPower;
```

File: src/main/java/com/teamcautionrobotics/robot/Robot.java [hunk 5]
```diff
@@ -25,131 +26,154 @@
...
+
+    // These are for the Expander Hatch mechanism
+    boolean reacherButtonPressed = false;
+    boolean grabberButtonPressed = false;
+
+    private final boolean usingVelcroHatch = true;
+    private static final String kDefaultAuto = "Default";
+    private static final String kCustomAuto = "My Auto";
+    private String m_autoSelected;
+    private final SendableChooser<String> m_chooser = new SendableChooser<>();
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/Robot.java:55]: I thought we deleted this in #16.

File: src/main/java/com/teamcautionrobotics/robot/Robot.java [hunk 6]
```diff
@@ -25,131 +26,154 @@
...
+        // pneumatic ports are not finalized
+        velcroHatch = new VelcroHatch(2, 1);
+        expanderHatch = new ExpanderHatch(3, 4);
+        driverLeft = new EnhancedJoystick(0);
+        driverRight = new EnhancedJoystick(1);
+        manipulator = new Gamepad(2);
+        aimingLights = new AimingLights(0, 1);
+        driveBase = new DriveBase(0, 1, 0, 1, 2, 3);
+        CameraServer.getInstance().startAutomaticCapture(0);
+        CameraServer.getInstance().startAutomaticCapture(1);
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/Robot.java]: Some grouping here might be nice.

File: src/main/java/com/teamcautionrobotics/robot/Robot.java [hunk 7]
```diff
@@ -25,131 +26,154 @@
...
+     * <p>
+     * This runs after the mode specific periodic functions, but before LiveWindow
+     * and SmartDashboard integrated updating.
+     */
+    @Override
+    public void robotPeriodic() {
+        armPower = .5 + .5 * manipulator.getAxis(Axis.LEFT_TRIGGER);
+        velcroHatch.rotate(armPower);
+
+        // if B is pressed, deploy hatch pneumatics
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/Robot.java]: Errant comment

File: src/main/java/com/teamcautionrobotics/robot/Robot.java [hunk 8]
```diff
@@ -25,131 +26,154 @@
...
+     * <p>
+     * You can add additional auto modes by adding additional comparisons to the
+     * switch structure below with additional strings. If using the SendableChooser
+     * make sure to add them to the chooser code above as well.
+     */
+    @Override
+    public void autonomousInit() {
+        m_autoSelected = m_chooser.getSelected();
+        // m_autoSelected = SmartDashboard.getString("Auto Selector", kDefaultAuto);
+        System.out.println("Auto selected: " + m_autoSelected);
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/Robot.java:151]: Why is this here? I thought this was banished from our codedom.

File: src/main/java/com/teamcautionrobotics/robot/Robot.java [hunk 9]
```diff
@@ -25,131 +26,154 @@
...
+        driveRightCommand = driverRight.getY();
+
+        aimingLights.set(driverLeft.getRawButton(2));
+
+        if (usingVelcroHatch) {
+            if (manipulator.getButton(Button.B)) {
+                velcroHatch.deploy(manipulator.getButton(Button.B));
+                deployButtonPressed = true;
+            } else {
+                if (deployButtonPressed) {
```
Comment (docprofsky) [inline src/main/java/com/teamcautionrobotics/robot/Robot.java:105]: Do we want to require releasing the button before the timer starts?

**Review by docprofsky — CHANGES_REQUESTED** [review]
There seems to be some autonomous code that snuck in. Everything else looks good.

File: src/main/java/com/teamcautionrobotics/robot/Robot.java (hunk 5 above)
Comment (TaeSeanDo) [reply src/main/java/com/teamcautionrobotics/robot/Robot.java:55]: I was planning on fixing this when resolving merge conflicts.

File: src/main/java/com/teamcautionrobotics/robot/AimingLights.java (hunk 1 above)
Comment (Ben1492) [inline src/main/java/com/teamcautionrobotics/robot/AimingLights.java]: Can we name the method for this class  "toggle" instead of "set," since that is what it is doing?

Comment (TaeSeanDo) [reply src/main/java/com/teamcautionrobotics/robot/AimingLights.java]: I think set is a better name because when you call the method, it does not necessarily change state, e.g. calling aimingLights.set(true) when the lights are already on.

File: src/main/java/com/teamcautionrobotics/robot/Robot.java (hunk 9 above)
Comment (TaeSeanDo) [reply src/main/java/com/teamcautionrobotics/robot/Robot.java:105]: We already do.

File: src/main/java/com/teamcautionrobotics/robot/Robot.java (hunk 8 above)
Comment (TaeSeanDo) [reply src/main/java/com/teamcautionrobotics/robot/Robot.java:151]: Same reason: branched from an out-of-date master.

File: src/main/java/com/teamcautionrobotics/robot/Robot.java (hunk 9 above)
Comment (docprofsky) [reply src/main/java/com/teamcautionrobotics/robot/Robot.java:105]: Is this correct behavior?

File: src/main/java/com/teamcautionrobotics/robot/Robot.java (hunk 5 above)
Comment (docprofsky) [reply src/main/java/com/teamcautionrobotics/robot/Robot.java:55]: That seems good.

File: src/main/java/com/teamcautionrobotics/robot/Robot.java (hunk 9 above)
Comment (TaeSeanDo) [reply src/main/java/com/teamcautionrobotics/robot/Robot.java:105]: Yes, and it is tested.

**Review by docprofsky — APPROVED** [review]
